CorsixTH/Lua/graphics.lua

```diff
@@ -254,7 +262,7 @@ function Graphics:loadGhost(dir, name, index)
   return cached:sub(index * 256 + 1, index * 256 + 256)
 end
 
-function Graphics:loadRaw(name, width, height, dir, paldir, pal)
+function Graphics:loadRaw(name, width, height, dir, paldir, pal, transparent_255)
```

Comment:
I don't like this method signature at all. I want Graphics:loadRaw(dir, name, width, heigh, palette) - but I can't change it without breaking compatibility because of the load table.

Reply:
That deserves a further explanation:

Loading the same palette more than once with different flags doesn't work as expected, they are cached by name only; so when I was simply adding the transparent_255 flag to loadPalette, it was still getting loaded without transparency because there was a loadRaw earlier - and the loadRaw didn't even reference a palette, it automatically loads a palette matching the name of the bitmap.

I'm unimpressed.